### PR TITLE
Add/document is_disabled, is_listed, is_source_public to add-ons API

### PR DESCRIPTION
**Pull request by diox**
Also remove some workarounds in ES indexation code that are no longer needed.

### CODE REVIEW

File: src/olympia/addons/tests/test_serializers.py [hunk 1]
```diff
@@ -153,6 +153,24 @@ def test_basic(self):
...
+        result = self.serialize()
+
+        assert result['is_listed'] is False
+
+    def test_is_source_public(self):
+        self.addon = addon_factory(view_source=True)
+        result = self.serialize()
+
+        assert result['is_source_public'] is True
+
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: I love tests that check that it's the other way around before and then check that the value changed the way we expected it. But there are a few other tests that might check that so I guess it should be okay.

**Comment by EnTeQuAk** [comment]
r+wc (probably not so important comment)

File: src/olympia/addons/tests/test_serializers.py (hunk 1 above)
Comment (diox) [inline src/olympia/addons/tests/test_serializers.py]: They are in `test_basic()` above.

Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: Indeed I thought so

**Pull request merged**